### PR TITLE
[oneMKL] Interface lapack routines

### PR DESCRIPTION
**Pull request by amontoison**
@kballeda @pengtu 
I wrapped additional LAPACK routines but I have some errors during the compilation.
Do you also have this issue?

Update:  The culprit was my local Intel compiler v2023.2.0 that was used. I'm able to compile with the compile installed by Conda.

### CODE REVIEW

**Comment by amontoison** [comment]
@kballeda I added more LAPACK routines and updated `support.toml` for the wrappers of your PR #348.

**Comment by kballeda** [comment]
> @kballeda I added more LAPACK routines and updated `support.toml` for the wrappers of your PR #348.

Thanks for adding ormqr, unmqr, orgqr, ungqr primitives to the list. It's good to have tests for these primitives in test/onemkl.jl.

**Comment by amontoison** [comment]
I will add some tests asap. What can we do to fix CI?

**Comment by kballeda** [comment]
> I will add some tests asap. What can we do to fix CI?

@amontoison  https://github.com/JuliaGPU/oneAPI.jl/issues/357 - Discussion on this topic

**Comment by amontoison** [comment]
@pengtu @kballeda
I have a segmentation fault when I try to call a LAPACK routine like `getrf!`.
Do you have an idea of what is the issue?
Do I allocate and use correctly the buffer?

File: lib/mkl/wrappers_lapack.jl [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+                             (:onemklZpotrf_scratchpad_size, :onemklZpotrf, :ComplexF64))
+    @eval begin
+        function potrf!(uplo::Char,
+                        A::oneStridedMatrix{$elty})
+            chkuplo(uplo)
+            n = checksquare(A)
+            lda = max(1, stride(A, 2))
+
+            queue = global_queue(context(A), device(A))
+            scratchpad_size = $bname(device_queue, uplo, n, lda)
```
Comment (amontoison) [inline lib/mkl/wrappers_lapack.jl:14]:
```suggestion
            scratchpad_size = $bname(sycl_queue(queue), uplo, n, lda)
```

**Comment by maleadt** [comment]
I added you to the oneAPI.jl team; feel free to merge whenever this is ready and passes CI.

**Comment by amontoison** [comment]
> I added you to the oneAPI.jl team; feel free to merge whenever this is ready and passes CI.

Thanks @maleadt!

**Pull request merged**